homeassistant/components/ping/binary_sensor.py

```diff
@@ -131,23 +133,50 @@ def __init__(self, host, count):
                 self._ip_address,
             ]
 
-    def ping(self):
+    async def async_ping(self):
         """Send ICMP echo request and return details if success."""
-        pinger = subprocess.Popen(
-            self._ping_cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+        pinger = await asyncio.create_subprocess_exec(
+            *self._ping_cmd,
+            stdin=None,
+            stdout=asyncio.subprocess.PIPE,
+            stderr=asyncio.subprocess.PIPE,
         )
         try:
-            out = pinger.communicate(timeout=self._count + PING_TIMEOUT)
-            _LOGGER.debug("Output is %s", str(out))
+            out_data, out_error = await pinger.communicate(
```

Comment:
Async subprocess has no timeout. Did you test the code?
https://docs.python.org/3/library/asyncio-subprocess.html#asyncio.asyncio.subprocess.Process.communicate

you need use async_timeout and also kill the process on timeout and wait for the process.

Reply:
@pvizeli can't I use `asyncio.wait_for`? Similar to this: https://stackoverflow.com/a/29756841/965722
I can't find a proper example of using `async-timeout`
After the last change, the code works fine. However, I'm not sure how to properly check for timeouts.